seacrowd/sea_datasets/indonesian_madurese_bible_translation/indonesian_madurese_bible_translation.py

```diff
@@ -0,0 +1,238 @@
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "src": datasets.Value("string"),
+                    "tgt": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+        data_dir = os.path.join(data_dir, "Bahasa Madura Corpus Dataset/Indonesian-Madurese Corpus")
+        all_raw_path = [data_dir + "/" + item for item in os.listdir(data_dir)]
+        all_path = []
+        id = 0
+        for raw_path in all_raw_path:
+            if "txt" in raw_path:
+                all_path.append(raw_path)
+        all_data = []
+        for path in all_path:
+            data = self._read_txt(path)
+            for line in data:
+                if line != "\n":
+                    all_data.append({"src": line.split("\t")[0], "tgt": line.split("\t")[1], "id": id})
+                id += 1
+        self._write_jsonl(data_dir + "/train.jsonl", all_data)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # Whatever you put in gen_kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "train.jsonl"),
+                    "split": "train",
+                },
+            )
+        ]
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+
+    # TODO: change the args of this function to match the keys in `gen_kwargs`. You may add any necessary kwargs.
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        if self.config.schema == "source":
+            i = 0
+            with jsonlines.open(filepath) as f:
+                for each_data in f.iter():
+                    ex = {
+                        "id": each_data["id"],
+                        "src": each_data["src"],
+                        "tgt": each_data["tgt"],
+                    }
+                    yield i, ex
+                    i += 1
+
+        elif self.config.schema == "seacrowd_t2t":
+            i = 0
+            with jsonlines.open(filepath) as f:
+                for each_data in f.iter():
+                    ex = {"id": each_data["id"], "text_1": each_data["src"], "text_2": each_data["tgt"], "text_1_name": "ind", "text_2_name": "mad"}
```

Comment:
Could you please `.strip()` the texts to remove leading and/or trailing whitespaces?

Reply:
Done